peartree/parallel.py

```diff
@@ -74,9 +74,19 @@ def generate_route_costs(self, route_id: str):
                      'departure_time']
         trips_and_stop_times = trips_and_stop_times.sort_values(sort_list)
 
+        # Check direction_id column value before 
```

Comment:
Per our discussion offline - the advantage to adding the handling here is that it is within a single route. This means that we do not end up tossing direction id if a specific route happens to have all direction id rows filled in. It is possible that a route operator has direction id for one route but not another.

Reply:
comments added. thanks!